internal/pkg/helper/helper.go

```diff
@@ -0,0 +1,68 @@
+package handler
+
+import (
+	"bytes"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+)
+
+// ConvertToEnvVarName converts the given text into a usable env var
+// removing any special chars with '_' and transforming text to upper case
+func ConvertToEnvVarName(text string) string {
+	var buffer bytes.Buffer
+	upper := strings.ToUpper(text)
+	lastCharValid := false
+	for i := 0; i < len(upper); i++ {
+		ch := upper[i]
+		if (ch >= 'A' && ch <= 'Z') || (ch >= '0' && ch <= '9') {
+			buffer.WriteString(string(ch))
+			lastCharValid = true
+		} else {
+			if lastCharValid {
+				buffer.WriteString("_")
+			}
+			lastCharValid = false
+		}
+	}
+	return buffer.String()
+}
+
+// ConvertConfigmapToSHA generates SHA for configmap data
+func ConvertConfigmapToSHA(cm *v1.ConfigMap) string {
```

Comment:
ConvertConfigMapToSHA and ConvertSecretToSHA are identical except for 1 small thing.
Responsibility of converting some string to SHA should be separate and in a separate package. This helper doesn't really make sense since it isn't clear of what it is supposed to do. Create meaningful packages and try to keep things decoupled from kubernetes objects at the bottom level